tools/wpt/revlist.py

```diff
@@ -0,0 +1,299 @@
+                "(default: %s seconds)." % DEFAULT_EPOCH_THESHOLD,
+        "parser": lambda **kwargs: int(kwargs.get("epoch_threshold",
+                                                  DEFAULT_EPOCH_THESHOLD)),
+        "type": int
+    },
+
+    "epoch_step": {
+        "name": "epoch-step",
+        "default": DEFAULT_EPOCH_STEP,
+        "help": "Regular interval of seconds used to get the tagged revision. "
+                "three-hourly, six-hourly, eight-hourly, twelve-hourly, "
+                "daily, weekly or 3600, 7200, 10800 ...",
+        "parser": parser_epoch_step
+    },
+    "head_revision": {
+        "name": "head-revision",
+        "default": DEFAULT_HEAD_REVISION,
+        "help": "Head revision (default: %s)" % DEFAULT_HEAD_REVISION,
+        "parser": lambda **kwargs: kwargs.get("head_revision",
+                                              DEFAULT_HEAD_REVISION),
+    },
+    "log_level": {
+        "name": "log-level",
+        "default": DEFAULT_LOG_LEVEL,
+        "help": "Debug log level. For example: DEBUG, INFO, WARNING, ERROR, "
+                "CRITICAL or an integer value like 10, 20 ",
+        "parser": parser_debug_log_level
+    },
+    "num_revisions": {
+        "name": "num-revisions",
+        "default": DEFAULT_NUM_REVISIONS,
+        "help": "Number of revisions to get.",
+        "parser": parser_num_revisions
+    },
+    "verbose": {
+        "name": "verbose",
+        "default": DEFAULT_VERBOSE,
+        "help": "Show more information in the output",
+        "parser": lambda **kwargs: kwargs.get("verbose", DEFAULT_VERBOSE),
+        "type": bool
+    }
+}
+
+
+def get_git_cmd(repo_path):
+    # type: (bytes) -> Callable[..., Text]
+    """Create a function for invoking git commands as a subprocess."""
+    def git(cmd, *args):
+        # type: (Text, *Union[bytes, Text]) -> Text
+        arg_list = list(item.decode("utf8") if isinstance(item, bytes) else item for item in args)  # noqa E501
+        full_cmd = [u"git", cmd] + arg_list  # type: List[Text]
+        try:
+            logger.debug(" ".join(full_cmd))
+            return subprocess.check_output(full_cmd, cwd=repo_path).decode("utf8").strip()  # noqa E501
+        except subprocess.CalledProcessError as e:
+            logger.critical("Git command exited with status %i" % e.returncode)
+            logger.critical(e.output)
+            sys.exit(1)
+    return git
+
+
+def get_tagged_revisions(commit, min_age, skip=0, max_count=10):
+    # type: (bytes, int, int, int) -> Dict[..., Dict]
+    '''
+    Returns the tagged revisions indexed by the committer date.
+    '''
+    git = get_git_cmd(wpt_root)
+    command = [
+        "--no-walk",
+        "--skip=%s" % skip,
+        "--max-count=%s" % max_count,
+        "--min-age=%s" % min_age,
+        "--format=epoch:%ct:commit:%H:%D",
+        commit
+    ]
+    result = {}
+    for output_line in git("rev-list", *command).split("\n")[1::2]:
+        # print output_line
```

Comment:
I uncommented the to see what the the output being parsed looks like. Sample:
```
epoch:1570607217:commit:be08de67a0b21e800cc263e319aa4f17034bae53:tag: merge_pr_19560, origin/master, origin/HEAD, master
epoch:1570605657:commit:3c62b4e788a0f5e91a620cd2b97c14d4b64d9182:tag: merge_pr_19463
epoch:1570599487:commit:f44adb8c6eef6ae422f145e1a74d0b0c916f2a64:tag: merge_pr_19547
[... more of the same ...]
epoch:1570553109:commit:7fbe6fe5fe834bf28d9171bf92d90287186020e8:tag: merge_pr_19541
epoch:1570551066:commit:016aca0f8606b21284161f2772c461e521ce253b:tag: merge_pr_19457
epoch:1570551030:commit:3982e1b6a11e163be4469598421b0cbcf6ecab82:tag: merge_pr_19327
epoch:1570551030:commit:d7789eace2c576625633a913cf5f835aae68238e:
epoch:1570546541:commit:7a7a826cd32459b449cc92fba7135ce9039aeb6f:tag: merge_pr_19524
epoch:1570546067:commit:cdee611d3fe90fd05c3da262b81cf6b7c0414b9a:tag: merge_pr_19569
epoch:1570545719:commit:7771c4cda2aa570d0bb54d6b7b28e804f0e85e2e:tag: merge_pr_19571
```

In other words, description bits other than tags can be included, and commits without tags can also be included.

There is a similar piece of code for downloading a recent manifest:
https://github.com/web-platform-tests/wpt/blob/6e583d30af20cef1eae00927b353d525fe5ef7b2/tools/manifest/download.py#L52-L62

That code only needs the tag names, however, so I won't suggest trying to refactor+reuse it.

Instead, I'll suggest again to use `git for-each-ref 'refs/tags/merge_pr_*' --sort=-committerdate --format="%(refname:lstrip=2) %(objectname) %(committerdate:iso-strict)"`, possibly with `--count=10000` to limit the number of lines processed.

The ease of parsing that and knowing that you'll get exactly one tag per line is more important than the ability to list only tags reachable from HEAD, I think. Note that `--no-containts` could be used if that were important, but I say not for now :)